projects/jdwp/codegen/deserializer_generator.py

```diff
@@ -0,0 +1,35 @@
+from projects.jdwp.defs.schema import Command, Field, PrimitiveType
+
+
+def generate_deserialization_function(command: Command) -> str:
```

Comment:
We want to generate `serialize` and `parse` methods for all generated structs. Once that happens we will make them subclasses of [JDWPStruct](https://github.com/facebookexperimental/ExtendedAndroidTools/blob/main/projects/jdwp/runtime/jdwpstruct.py#L12). For that we need two methods for each struct:
- `def serialize(self, output: JDWPOutputStreamBase)`
- static `async def parse(input: JDWPInputStreamBase)`

In order to generate both methods we need to iterate through fields defined in the spec and read/write them. Example of parse for `__ClassPaths_reply`:
```
@staticmethod
async def parse(input: JDWPInputStreamBase) -> ClassPathsReply:
    baseDir: str = await input.read_string()

    classpaths: int = await input.read_int()

    classpathsArray: List[ClassPathsReplyClasspathsArrayElement] = []
    for _ in range(classpaths):
        classpathsArray.append(await ClassPathsReplyClasspathsArrayElement.parse(input))

    bootclasspaths = await input.read_int()
    
    bootclasspathsArray: List[ClassPathsReplyClasspathsArrayElement] = []
    for _ in range(bootclasspaths):
        bootclasspathsArray.append(await ClassPathsReplyBootclasspathsArrayElement.parse(input))

    return ClassPathsReply(
        baseDir=baseDir,
        classpathsArray=classpathsArray,
        bootclasspathsArray=bootclasspathsArray,
    )
```

Serialize method for the same struct could look like:
```
def serialize(self, output: JDWPOutputStreamBase):
    output.write_string(self.baseDir)

    output.write_int(len(self.classpathsArray))

    for element in self.classpathsArray:
        element.serialize(output)
        
    output.write_int(len(self.bootclasspathsArray))

    for element in self.bootclasspathsArray:
        element.serialize(output)
```

Tags of unions and lengths of arrays are interesting: they don't have corresponding fields in generated structs, but those values. Those values are used only to guide the parsing logic and need to be computed when serializing structs.

Reply:
We probably want to make `generate_deserialization_function` and `generate_serialization_function` methods of `StructGenerator` so that we can generate parsing and serializing functions as methods of corresponding structs.

Reply:
One more thing, [JDWPStruct.serialize](https://github.com/facebookexperimental/ExtendedAndroidTools/blob/main/projects/jdwp/runtime/jdwpstruct.py#L14) is an async method, but I will be changing it now. Writing to a stream is not async, so we can keep it simple.